src/network/protocol/messages/getblocks.zig

```diff
@@ -0,0 +1,186 @@
+const std = @import("std");
+const native_endian = @import("builtin").target.cpu.arch.endian();
+const protocol = @import("../lib.zig");
+
+const ServiceFlags = protocol.ServiceFlags;
+
+const Endian = std.builtin.Endian;
+const Sha256 = std.crypto.hash.sha2.Sha256;
+
+const CompactSizeUint = @import("bitcoin-primitives").types.CompatSizeUint;
+
+/// GetblocksMessage represents the "getblocks" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#getblocks
+pub const GetblocksMessage = struct {
+
+    version: i32,
+    hash_count: u64,
+    header_hashes: [] [32]u8,
+    stop_hash: [32]u8,
+
+    pub inline fn name() *const [12]u8 {
+        return protocol.CommandNames.GETBLOCKS ++ [_]u8{0} ** 5;
+    }
+
+    /// Returns the message checksum
+    ///
+    /// Computed as `Sha256(Sha256(self.serialize()))[0..4]`
+    pub fn checksum(self: *const GetblocksMessage) [4]u8 {
+        var digest: [32]u8 = undefined;
+        var hasher = Sha256.init(.{});
+        const writer = hasher.writer();
+        self.serializeToWriter(writer) catch unreachable; // Sha256.write is infaible
+        hasher.final(&digest);
+
+        Sha256.hash(&digest, &digest, .{});
+
+        return digest[0..4].*;
+    }
+
+    /// Free the `header_hashes`
+    pub fn deinit(self: GetblocksMessage, allocator: std.mem.Allocator) void {
+        if (self.header_hashes.len > 0) {
+            allocator.free(self.header_hashes);
+        }
```

Comment:
no need to check for len as zig allocators impl for free is a noop in case of empty array